### PR TITLE
Moengage-segment integration

### PR DESCRIPTION
**Pull request by shubham-engg-data**
This PR defines the streaming MoEngage destination on Segment.

Created a fresh new destination with two actions as currently at MoEngage, we are tracking the users as well their events and for that, we have created the two actions:

1. trackEvent: To track the user events
2. identifyUser: what are the users that need to add

At MoEngage we are supporting 3 regions i.e:
1.  **DC-01** ( us-east-1 )
2.  **DC-02** ( eu-central-1 )
3.  **DC-03** ( ap-south-1 )

and for that, we have separate endpoints for each of them that we define in this PR.

Defined all the input fields in both the actions(trackEvent, identifyUser) that we are currently using in our API. In perform function of each action defines the actual logic of generating the actual payload that MoEngage supports and calling of our integrations API.

## Testing

_Include any additional information about the testing you have completed to
ensure your changes behave as expected. For a speedy review, please check
any of the tasks you completed below during your testing._

- [x] Added [unit tests](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing) for new functionality
- [x] Tested end-to-end using the [local server](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing)
- [ ] [Segmenters] Tested in the staging environment

### CODE REVIEW

File: packages/destination-actions/src/destinations/actions-moengage/identifyUser/index.ts [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+      type: payload.type,
+      traits: payload.traits,
+      context:{}
+    }
+
+    if (!settings.api_id || !settings.api_key) {
+      throw new IntegrationError('Missing API ID or API KEY', 'Missing required field', 400)
+    }
+
+    if (payload.user_id) {
```
Comment (dlasky) [inline packages/destination-actions/src/destinations/actions-moengage/identifyUser/index.ts:93]: suggest removing the ifs as the 'event' will run through JSON.stringify which will strip any undefined values

File: packages/destination-actions/src/destinations/actions-moengage/index.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        required: true
+      },
+      api_key: {
+        label: 'Api Key',
+        description: 'Your Moengage API Key',
+        type: 'string',
+        required: true
+      },
+      region: {
+        label: 'Endpoint Region',
```
Comment (dlasky) [inline packages/destination-actions/src/destinations/actions-moengage/index.ts:28]: suggest making this required due to how our auth settings work currently in the UI

File: packages/destination-actions/src/destinations/actions-moengage/trackEvent/index.ts [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+    }
+
+    const event: any = {
+      type: payload.type,
+      event: payload.event,
+      context: {},
+      properties: payload.properties,
+    }
+
+    if (payload.user_id) {
```
Comment (dlasky) [inline packages/destination-actions/src/destinations/actions-moengage/trackEvent/index.ts:100]: suggest removing ifs similar to above comment

**Review by dlasky — CHANGES_REQUESTED** [review]
lgtm, a few minor nits!

File: packages/destination-actions/src/destinations/actions-moengage/identifyUser/index.ts (hunk 1 above)
Comment (shubham-engg-data) [reply packages/destination-actions/src/destinations/actions-moengage/identifyUser/index.ts]: Done

File: packages/destination-actions/src/destinations/actions-moengage/index.ts (hunk 2 above)
Comment (shubham-engg-data) [reply packages/destination-actions/src/destinations/actions-moengage/index.ts:28]: Done

File: packages/destination-actions/src/destinations/actions-moengage/trackEvent/index.ts (hunk 3 above)
Comment (shubham-engg-data) [reply packages/destination-actions/src/destinations/actions-moengage/trackEvent/index.ts]: Done

File: packages/destination-actions/src/destinations/actions-moengage/index.ts [hunk 4]
```diff
@@ -0,0 +1,67 @@
+import type { DestinationDefinition } from '@segment/actions-core'
+import type { Settings } from './generated-types'
+import trackEvent from './trackEvent'
+import identifyUser from './identifyUser'
+import { getEndpointByRegion } from './regional-endpoints'
+
+const destination: DestinationDefinition<Settings> = {
+  name: 'Moengage',
```
[the remark this answers is not in the document]
Comment (shubham-engg-data) [reply packages/destination-actions/src/destinations/actions-moengage/index.ts]: Done @nick-Ag , Thanks for the suggestion .

**Review by nick-Ag — APPROVED** [review]

**Pull request merged**